components/raftstore/src/store/fsm/apply.rs

```diff
@@ -2692,77 +2579,35 @@ where
     },
 }
 
-pub struct ApplyFsm<EK>
-where
-    EK: KvEngine,
-{
-    delegate: ApplyDelegate<EK>,
-    receiver: Receiver<Msg<EK>>,
-    mailbox: Option<BasicMailbox<ApplyFsm<EK>>>,
-}
-
 impl<EK> ApplyFsm<EK>
 where
     EK: KvEngine,
 {
-    fn from_peer<ER: RaftEngine>(
-        peer: &Peer<EK, ER>,
-    ) -> (LooseBoundedSender<Msg<EK>>, Box<ApplyFsm<EK>>) {
-        let reg = Registration::new(peer);
-        ApplyFsm::from_registration(reg)
-    }
-
-    fn from_registration(reg: Registration) -> (LooseBoundedSender<Msg<EK>>, Box<ApplyFsm<EK>>) {
-        let (tx, rx) = loose_bounded(usize::MAX);
-        let delegate = ApplyDelegate::from_registration(reg);
-        (
-            tx,
-            Box::new(ApplyFsm {
-                delegate,
-                receiver: rx,
-                mailbox: None,
-            }),
-        )
-    }
-
-    /// Handles peer registration. When a peer is created, it will register an apply delegate.
-    fn handle_registration(&mut self, reg: Registration) {
-        info!(
-            "re-register to apply delegates";
-            "region_id" => self.delegate.region_id(),
-            "peer_id" => self.delegate.id(),
-            "term" => reg.term
-        );
-        assert_eq!(self.delegate.id, reg.id);
-        self.delegate.term = reg.term;
-        self.delegate.clear_all_commands_as_stale();
-        self.delegate = ApplyDelegate::from_registration(reg);
-    }
-
     /// Handles apply tasks, and uses the apply delegate to handle the committed entries.
-    fn handle_apply<W: WriteBatch<EK>>(
+    async fn handle_apply<W: WriteBatch<EK>>(
         &mut self,
         apply_ctx: &mut ApplyContext<EK, W>,
         mut apply: Apply<EK::Snapshot>,
     ) {
-        if apply_ctx.timer.is_none() {
-            apply_ctx.timer = Some(Instant::now_coarse());
-        }
-
-        fail_point!("on_handle_apply_1003", self.delegate.id() == 1003, |_| {});
-        fail_point!("on_handle_apply_2", self.delegate.id() == 2, |_| {});
+        let start_apply = if apply_ctx.timer.is_none() {
+            Some(Instant::now_coarse())
+        } else {
+            None
+        };
```

Comment:
Quite confusing about how `apply_ctx.timer` is set and what it will observe.

Reply:
I use `apply_ctx.timer` to calculate duration of twice `flush`. It may be a little difficult to understand and I will think about refactoring this code.